### ISSUE TITLE
Cell cycle scoring function in Mayer et al 2018 paper 

### ISSUE THREAD

**Issue by yueqiw** [satijalab/seurat#614]
Hi,

It looks like the paper "Developmental diversification of cortical inhibitory interneurons" by Mayer et al, used a slightly different method to score cell cycle phases compared to Seurat. 
https://github.com/ChristophH/in-lineage/blob/master/R/lib.R 

I tried the cell cycle scoring function from the above source code on my dataset and got somewhat better-looking results. 1. Tighter distribution of non-cycling cells and better separation of cycling vs. non-cycling cells, especially on smaller dataset. 2. The ranges of S.Score and G2M Score are more similar to each other using the in-lineage code than Seurat).   

I'm wondering if the in-lineage source code is a better scoring function in general (e.g. for scoring any gene set)? And is it going to become an option for Seurat? Thanks!

**Comment by satijalab** [comment]
Thanks for the feedback. The two methods are nearly identical, but one uses log-normalized data and the other uses log-counts data. I would be surprised if there was a consistent difference between the two across datasets - have you tried both methods on our cell cycle tutorial datasets? If you or others see consistent differences, we could reconsider

**Comment by yueqiw** [comment]
Thanks! Another difference between the two methods seem to be how the final score is calculated: 

In the in-lineage code, there is a mean-sd scaling step:
```
return((goi.mean - apply(bg.mean, 1, mean)) / apply(bg.mean, 1, sd))
```
In Seurat, it's the difference between two scores:
```
genes.scores.use <- genes.scores - ctrl.scores
```

I haven't read the code very carefully, but this is what I have noticed so far. 

I didn't try it on the cell cycle tutorial dataset. The tutorial dataset seems to only contain progenitor cells (the ideal case). My data contains several progenitor cell types and also multiple differentiated cell types, so it's a bit more complicated. I tried on two of my datasets, and the in-lineage code gives more balanced S score and G2M score.  (And maybe because the complexity of cell types, cell cycle regression usually does not completely remove the effect of cell cycle or phase difference. It shifts the effects from top ~3 PC to something like the ~8th PC. So I ended up doing feature selection to remove cell cycle genes from HVGs. Now I'm interested in using the scoring function to look at other gene pathways.)
